core/src/main/java/tc/oc/pgm/filters/MultiFilterFunction.java

```diff
@@ -33,6 +36,13 @@ public MultiFilterFunction(Iterable<? extends Filter> filters) {
     this.filters = ImmutableList.copyOf(filters);
   }
 
+  @Override
+  public Collection<Class<? extends Event>> getRelevantEvents() {
+    List<Class<? extends Event>> allEvents = new ArrayList<>();
+    this.filters.forEach(f -> allEvents.addAll(f.getRelevantEvents()));
+    return ImmutableList.copyOf(allEvents);
```

Comment:
```suggestion
    return this.filters.stream().flatMap(f -> f.getRelevantEvents().stream())
            .collect(Collectors.collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
```
It hurts to see lambdas being used to "fill a collection" outside of the stream itself, when the stream is what should be collected into a list itself